youtube_transcript_api/test/test_api.py

```diff
@@ -176,15 +176,21 @@ def test_get_transcripts__continue_on_error(self):
 
         YouTubeTranscriptApi.get_transcripts(['video_id_1', 'video_id_2'], continue_after_error=True)
 
-        YouTubeTranscriptApi.get_transcript.assert_any_call(video_id_1, ('en',), None)
-        YouTubeTranscriptApi.get_transcript.assert_any_call(video_id_2, ('en',), None)
+        YouTubeTranscriptApi.get_transcript.assert_any_call(video_id_1, ('en',), None, None)
+        YouTubeTranscriptApi.get_transcript.assert_any_call(video_id_2, ('en',), None, None)
+   
+    def test_get_transcripts__check_cookies(self):
```

Comment:
This is not enough to get full coverage unfortunately. Since `get_transcript` is mocked, the `if cookies: ...` branch in `list_transcripts` never gets executed during testing.

Reply:
I can go more in depth for sure, however I have noticed an error with the testing suite. The calling of MagicMock() over the `get_transcripts()` is actually a global set, so after the first time it is called no subsequent test can actually test that call. I suspect that is why the `test_get_transcript__with_proxies` doesn't follow the previous `test_get_transcripts` naming convention even though it is testing `get_transcripts()`. This is done so alphabetically (which I think is how coverage runs) it can test `get_transcripts()` before setting and checking a mock of `get_transcript()`.

I think this should be resolved by moving away from `MagicMock` and using [patch()](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch) from the same library.  However, can I do that in this PR to get my testing working or do I need to open another PR?

Reply:
That is a very good point. I didn't think this through when implementing it like this, but mocking `get_transcript` on a module level is not desirable at all. As you pointed out the `MagicMock` should be patched for each test individually. We should definitely get that fixed.
`test_get_transcript__with_proxies` was part of a contribution and looking back at it I shouldn't have accepted it like that, since it doesn't really test if the requests are actually called with the proxy settings, as you have pointed out.